sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/zapwrapper/helper/ZapProductMessageHelper.java

```diff
@@ -58,14 +60,23 @@ public void writeProductError(ZapWrapperRuntimeException zapWrapperRuntimeExcept
         }
     }
 
-    public void writeUserMessagesWithScannedURLs(List<String> urls) {
-        for (String url : urls) {
+    public void writeUserMessagesWithDetectedURLs(List<Map<String, ApiResponse>> results) {
+        // The map looks like this: {processed=true, statusReason=OK, method=GET,
+        // reasonNotProcessed=, messageId=6, url=http://example.com, statusCode=200}
+        for (Map<String, ApiResponse> result : results) {
+            String url = result.get("url").toString();
             // robots.txt and sitemap.xml always appear inside the sites tree even if they
             // are not available. Because of this it is skipped here.
             if (url.contains("robots.txt") || url.contains("sitemap.xml")) {
                 continue;
             }
-            writeSingleProductMessage(new SecHubMessage(SecHubMessageType.INFO, "Detect url to scan: " + url));
+            String statusCode = result.get("statusCode").toString();
```

Comment:
again: is this ensured that it cannot be null? It's just a map...
Maybe the easiest way would be to provide a fetchSafe(map, $key, $defaultFallback) method to implement and to use...